test/mocks/network/mocks.h

```diff
@@ -246,6 +247,46 @@ class MockListenSocket : public Socket {
   bool socket_is_open_ = true;
 };
 
+class MockSocket : public Socket {
+public:
+  MockSocket()
+      : io_handle_(std::make_unique<MockIoHandle>()),
+        options_(std::make_shared<std::vector<OptionConstSharedPtr>>()){};
+  ~MockSocket() override = default;
```

Comment:
Move constructor/destructor to cc file per other mocks and per gmock compile speed guidelines.

Reply:
Ok. I will move all implementations to cc file not header file.